jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ByteBufferPool.java

```diff
@@ -158,6 +158,14 @@ public RetainableByteBuffer acquire()
         }
     }
 
+    /**
+     * A marker interface for classes that expose a pre-sized buffer pool.
+     */
+    interface Holder
+    {
+        ByteBufferPool.Sized getSizedByteBufferPool();
+    }
```

Comment:
This is a little better... but I still don't like that wrapping can hide it.

We have adopted a wrapper heavy approach, so marker interfaces do not work well, unless we can unwrap all the layers looking for the marker.

Perhaps we need a Content.Sink.Wrapper class and then a static method that looks through the Sink and any wrappers looking for a Holder?

Reply:
The other option is to just make getByteBufferPool() a method on both Content.Sink and Content.Source

Reply:
This issue a way more complex than I anticipated, and since the best is the enemy of the good, I'm giving up and passing the sized pool as a parameter to `writeTo()`.

We could eventually revisit this later on if we find the time, but I don't want to block this PR on this matter.